src/main/java/com/bookMyShow/bookMyShow/controllers/SeatController.java

```diff
@@ -8,33 +8,47 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.util.ArrayList;
+import java.util.List;
+
 @RestController
 @RequestMapping(ApiConstant.API_V1_VERSIONING + SeatConstant.SEAT_BASE_URL)
 public class SeatController {
     @Autowired
-    SeatService seatService;
+    private SeatService seatService;
 
+    @PreAuthorize("hasAuthority('PRIVILEGE_THEATRE_CREATE')")
     @PostMapping
-    public ResponseEntity<ResponseFormatDto> save(@RequestBody SeatRequestDto seatRequestDto) {
-        Seat seat = seatService.save(seatRequestDto.getSeatNumber(),seatRequestDto.getSeatType(),seatRequestDto.getAuditoriumId());
-
-        SeatResponseDto seatResponseDto = SeatResponseDto.builder()
-                .seatNumber(seat.getSeatNumber())
-                .seatType(seat.getSeatType())
-                .id(seat.getId())
-                .auditoriumId(seat.getAuditorium().getId())
-                .build();
+    public ResponseEntity<ResponseFormatDto> createSeat(@RequestBody SeatRequestDto seatRequestDto) {
```

Comment:
Instead of taking all seats - lets just take rownum and colNum to get seats dynamically.